doc/lua_api.txt

```diff
@@ -6911,16 +6925,63 @@ of the `${k}` syntax in formspecs is not deprecated.
 * `set_float(key, value)`
 * `get_float(key)`: Returns `0` if key not present.
 * `get_keys()`: returns a list of all keys in the metadata.
-* `to_table()`: returns `nil` or a table with keys:
-    * `fields`: key-value storage
-    * `inventory`: `{list1 = {}, ...}}` (NodeMetaRef only)
-* `from_table(nil or {})`
-    * Any non-table value will clear the metadata
-    * See [Node Metadata] for an example
-    * returns `true` on success
+* `to_table()`:
+    * Returns metadata as a metadata table (see below), if possible
+    * Returns `nil` on failure
+* `from_table(data)`
+    * Imports metadata from a metadata table
+    * If `data` is a metadata table (see below), the metadata it represents
+      will replace all metadata of this MetaDataRef object
+    * Any non-table value for `data` will clear all metadata
+    * Returns `true` on success
 * `equals(other)`
     * returns `true` if this metadata has the same key-value pairs as `other`
 
+### Metadata tables
+
+A MetaDataRef can be converted to and from a metadata table,
+which is metadata represented in a table form. It is used
+to serialize and deserialize metadata.
+The methods `from_table` and `to_table` use metadata tables, as
+well as a few callback functions.
```

Comment:
What do you mean by "a few callback functions"? I do not understand what you want to say with the second part of this sentence.